### DISCUSSION TITLE
Linking against vcpkg feature library

### DISCUSSION THREAD

**Discussion by LittleTreeTrunk** [microsoft/vcpkg#40148] [Q&A] [answered]
I am using tiff library in my Project (Visual Studio, CMake, vcpkg in manifest mode). We are using the cxx (C++ version of tiff library) **feature** also. my vcpkg.json file contains the following "dependencies" to incorporate the tiff library.

`{ "name": "tiff", "features":["cxx"]}`

The 'usage' file in vcpkg specifies that tiff is to be used like the following
```

find_package(TIFF REQUIRED)
    target_link_libraries(main PRIVATE TIFF::TIFF)
``` 

In the project the package is found by giving the following command in the CMake file

`find_package(TIFF REQUIRED)`

Then it is linked by the following command

`target_link_libraries(myexe PRIVATE TIFF::TIFF)`

The project build gives me error related to tiffxx linking. 

May I know the right wat to link against 'tiffxx' through CMake?

**Comment by dg0yt** [comment, marked as answer]
`TIFF::TIFF` is the widely used target available via the Find module but not the C++ binding. You might use plain `find_library` for `[lib]tiffxx`. You will probably need to setup an imported target which also links `TIFF::TIFF`.

There should be exported CMake config (in the future). It is not yet installed by the port because it still lacks an exported `TIFF::TIFF`. Cf. https://gitlab.com/libtiff/libtiff/-/merge_requests/496.

**Reply by LittleTreeTrunk** [reply]
Thank you very much for the help.

I was going through the vcpkg documentation to understand how to link against an additional feature used.
If everything was correct in the case of tifxx, may I know how I would have linked it in the "right way" in my application?
